librad/src/git/storage.rs

```diff
@@ -385,6 +385,32 @@ impl<S: Clone> Storage<S> {
         Ok(Refs::from(signed))
     }
 
+    /// Get the names of the references that live under
+    /// `namespaces/<urn-id>/refs/heads/*`.
+    pub fn refs_heads(&self, urn: &RadUrn) -> Result<Vec<String>, Error> {
+        self.refs_heads_of(urn, None)
+    }
+
+    /// Get the names of the references that live under
+    /// `namspaces/<urn-id>/refs/remotes/<peer-id>refs/heads/*`.
+    pub fn refs_heads_of<P>(&self, urn: &RadUrn, peer: P) -> Result<Vec<String>, Error>
+    where
+        P: Into<Option<PeerId>>,
+    {
+        let heads = Reference {
+            namespace: urn.id.clone(),
+            remote: peer.into(),
+            category: RefsCategory::Heads,
+            name: "*".to_string(),
+        };
+
+        let mut names = vec![];
```

Comment:
Why allocate a `Vec` here? You can return the iterator if you do:

```rust
pub fn refs_heads_of<'a, P>(&'a self, urn: &RadUrn, peer: P) -> Result<impl Iterator<Item = Result<String, Error> + 'a, Error>
```

Reply:
I was running into lifetime issues but I think you're right it could work if I call `iter`

Reply:
Arrghhh, ya I'm still running into an issue that the `refs` that we get from `References::from_globs` will be dropped and its lifetime is tied to `ReferenceNames`.

But I suppose we can return `References` instead :)

Reply:
Look at `all_metadata` -- the opaque iterator needs to have the same lifetime as `&self` (via `+ 'a`). But yeah, a concrete type would also be fine.

Reply:
I think that would work if I wanted to return `Iterator<Item = Result<Reference<'a>, Error>>` but because I call `names` on the temporary value, returned by `References::from_globs`, it says that this will be dropped. I had the `+ 'a` on the `Iterator`, but it didn't seem to help.